src/libYARP_OS/include/yarp/os/impl/Terminal.h

```diff
@@ -0,0 +1,186 @@
+#include <yarp/os/impl/PlatformUnistd.h>
+#include <yarp/os/impl/PlatformStdio.h>
+
+#include <yarp/os/Port.h>
+#include <yarp/os/Bottle.h>
+#include <yarp/os/Vocab.h>
+
+#include <cstdio>
+#include <cstring>
+
+namespace yarp {
+namespace os {
+namespace impl {
+
+#ifdef WITH_LIBEDIT
+    #include <editline/readline.h>
+    #include <vector>
+    static std::vector<std::string> commands;
+    static yarp::os::Port* rpcHelpPort = nullptr;
+    static bool commandListInitialized = false;
+    static char* szLine = (char*)nullptr;
+    static bool readlineEOF=false;
+
+    static char* dupstr(char* s)
```

Comment:
These function should not be declared static in a .h file...
Also please move the implementation in a .cpp file